src/test/java/ua_parser/CachingParserTest.java

```diff
@@ -84,4 +92,19 @@ public void testCachedReplacementQuoting() throws Exception {
     super.testReplacementQuoting();
   }
 
+  public void parseAll() {
+    String agentString1 = "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; fr; rv:1.9.1.5) Gecko/20091102 Firefox/3.5.5,gzip(gfe),gzip(gfe)";
+    String agentString2 = "Mozilla/5.0 (iPhone; CPU iPhone OS 5_1_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9B206 Safari/7534.48.3";
+
+    Client expected1 = new Client(new UserAgent("Firefox", "3", "5", "5"),
+                                  new OS("Mac OS X", "10", "4", null, null),
+                                  new Device("Mac", "Apple", "Mac"));
+    Client expected2 = new Client(new UserAgent("Mobile Safari", "5", "1", null),
+                                  new OS("iOS", "5", "1", "1", null),
+                                  new Device("iPhone", "Apple", "iPhone"));
+
+    assertThat(parser.parse(agentString1), is(expected1));
+    assertThat(parser.parse(agentString2), is(expected2));
+  }
```

Comment:
[these](https://github.com/mbrtargeting/popper/blob/master/popper/segmem/data_parser.py#L224) are the features used from the python side - are we sure that this is going to translate 100% to the same set of features?

Reply:
I only moved that test :smile: It has been updated for the new model though, the more recent revision of `uap-core` can pick up that it's a Mac. As you can see in the `LegacyParserTest`, it has been detected as `Other` previously.